docs, config map: add small howto for ipvlan setup

borkmann opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6941)
<!-- Reviewable:end -->

borkmann committed:
docs: add note to minikube gsg for --kubernetes-version

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

borkmann committed:
docs: add initial networking gsg for ipvlan

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

joestringer commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,68 @@
+
+.. code:: bash
+
+    curl -LO https://raw.githubusercontent.com/cilium/cilium/1.4.0/examples/kubernetes/1.13/cilium.yaml
```
Please use `.. parsed-literal::` and `\ |SCM_WEB| \` (grep around for SCM_WEB and you'll see some examples).

joestringer commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,68 @@
+    - --datapath-mode=ipvlan
+    - --ipvlan-master-device=eth0
+    - --tunnel=disabled
+    - --install-iptables-rules=false
```
We should be able to do these with configmap options hmm? Simplify it a bit?

borkmann replied:
Ok, I'll add a paragraph on how to integrate into config map. Looking at other doc bits, looks like convention is to mention both.

joestringer replied:
With #6648 we should have no excuse to always do things the configmap way, but for v1.4 we need extra boilerplate in the daemon k8s YAMLs for it, similar to commits like this one which isn't yet in the codebase: https://github.com/mrostecki/cilium/commit/0e7cf76651a15be28d7729a38732f16b3ed799f8 .

borkmann replied:
Thanks for the pointers, will adapt the doc in similar way to your latter reference as it's needed in 1.4.

joestringer commented on Documentation/gettingstarted/minikube.rst:
```diff
@@ -33,6 +33,11 @@ Install kubectl & minikube
 
+Note that in case of installing Cilium for a specific Kubernetes version, the
+`--kubernetes-version vx.y.z` parameter can be appended to the `minikube start`
+command for bootstrapping the local cluster. By default, minikube will install
```
Single backticks in sphinx is a link. Double means a literal, you want double in this paragraph.

brb commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,112 @@
+It is also required to specify the ipvlan master device which typically
+points to a networking device that is facing the external network. This
+is done through setting ``ipvlan-master-device`` to the name of of the
+networking device such as ``"eth0"``, for example.
```
Maybe it's worth mentioning that the device name should be the same on each host.

brb commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,112 @@
+::
+
+  datapath-mode: "ipvlan"
+  ipvlan-master-device: "bond0"
```
To be consistent with the first paragraph, I would `s/bond0/eth0`.

borkmann committed:
cilium, ipvlan: add config map support to templates

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

borkmann committed:
cilium, ipvlan: add config map support to examples

Auto-generated via:

CILIUM_VERSION=latest make -C examples/kubernetes clean all

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

aanm commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,116 @@
+points to a networking device that is facing the external network. This
+is done through setting ``ipvlan-master-device`` to the name of the
+networking device such as ``"eth0"`` or ``"bond0"``, for example. For
+deployment, this device must have consistent naming on each host.
```
Replace this last sentence with something as: `Be aware this ConfigMap will be used by all nodes so it is required this device name is consistent on all nodes where you will deploy Cilium`

aanm commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,116 @@
+deployment, this device must have consistent naming on each host.
+
+The ipvlan datapath only supports direct routing mode, therefore
+tunneling is disabled through setting ``tunnel`` to ``"disabled"``.
```
Automatically? Do I, as a user, need to perform any action in the `tunnel` option?

borkmann replied:
Right now it needs to be disabled as provided in the example here as we might add a tunnel option still at a later point in time.

borkmann replied:
Will clarify this better.

aanm commented on Documentation/gettingstarted/ipvlan.rst:
```diff
@@ -0,0 +1,116 @@
+    NAME                READY     STATUS    RESTARTS   AGE
+    cilium-crf7f        1/1       Running   0          10m
+
+In order for L3S mode to work correctly, a kernel with the following fix
```
I think this warning / node should be when you are explaining that L3S is available. It's frustrating for a user to follow a tutorial, in this case with L3S enabled and then realize that he has to start over because he does not have the necessary requirements.

borkmann replied:
Yep will do, unfortunately there is no reasonable auto-detection for this ipvlan internal, but this fix is propagated to all stable kernels so at some point folks won't need to care. But will move this earlier.
